inc/app_replacer.php

```diff
@@ -212,8 +209,7 @@ public static function possible_tag_flags() {
 	 * @return array
 	 */
 	public static function possible_data_ignore_flags() {
-
-		if ( ! empty( self::$ignore_data_opt_attribute ) && is_array( self::$ignore_data_opt_attribute ) ) {
+		if ( ! empty( self::$ignore_data_opt_attribute ) ) {
```

Comment:
We also need the old condition, for the `null` logic to work as expected. 

For `null` we need the condition to be false, which without the `is_array` will be `true`.

Reply:
`! empty( null )` will also return the false, let me know if I'm getting something wrong here.

Reply:
^ @GrigoreMihai

Reply:
You are right @HardeepAsrani , sorry about the confusion I just followed how I remembered the logic should work. 

It appears it was not working as expected even before this pr, since `! empty( [] )` will also return false (unless I'm missing something) . 

If we just want to let it work how it did until now you are right there are no changes needed the `!empty` is enough. 

Since we found the mistake I think it will be a good idea to also fix it now as it's pretty small change. (just a suggestion, if you wish you can also leave it for later) 

I think changing to : `null !== self::$ignore_data_opt_attribute && is_array( self::$ignore_data_opt_attribute )` will make it work how it should: 

 - [ ] for `null` enter the setup flow
 - [ ] for `[]` return it

Reply:
I'll also add this.

Reply:
Before doing that, I think we can just make the condition to be:

`null !== self::$possible_src_attributes`

because the default is null, so it will always be an array if it's not null. I'm sure having the is_array check will provoke PHPStan. What do you think? @GrigoreMihai

Reply:
Can't think of a downside of what you said. 

Should work unless it gets setup to some non-array value somewhere else by mistake.

Reply:
That's the point of having PHPStan to find such cases and fixing them, so I think QA should most likely find them.